libsolidity/ast/AST.h

```diff
@@ -268,6 +268,12 @@ class Declaration: public ASTNode, public Scopable
 	/// @returns true if this is a declaration of a parameter of an event.
 	bool isEventParameter() const;
 
+	/// @returns true if the name can be used as is.
+	/// Usually the name alone can be used to refer to the corresponding entity.
+	/// But, for example, struct member names or enum member names require a prefix.
+	/// Another example is event parameter names, which do not participate in any proper scope.
+	bool isVisibleAsUnqualifiedName() const;
```

Comment:
I think this should state more explicitly that `false` means that the name can **never** be used as is. Originally I wanted to comment that there must be something wrong here because this should depend on the scope where the name is used. Only after reading other comments I understood that this is meant only for cases where we know for certain. I.e. `true` really means that the name could be visible in some scope.

```suggestion
	/// @returns false if the declaration can never be referenced without being qualified with a scope.
	/// Usually the name alone can be used to refer to the corresponding entity.
	/// But, for example, struct member names or enum member names always require a prefix.
	/// Another example is event parameter names, which do not participate in any proper scope.
	bool isVisibleAsUnqualifiedName() const;
```